apps/teams-test-app/src/components/ThirdPatryCookies.tsx

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable @microsoft/sdl/no-cookies */
+import React, { ReactElement } from 'react';
+
+import { ApiWithoutInput } from './utils';
+import { ModuleWrapper } from './utils/ModuleWrapper';
+
+const SetCookie = (): React.ReactElement => {
+  const setCookies = async (): Promise<string> => {
+    try {
+      const access = await requestCookieAccess();
+      if (!access) {
+        return 'Permission denied. Either blocked by user or browser';
+      }
+
+      document.cookie = 'testCookies=testvalue; path=/; Secure; SameSite=None';
+      return document.cookie.includes('testCookies=testvalue') ? 'true' : 'false';
+    } catch (err) {
+      console.error('Error setting cookies:', err);
+      return 'false';
+    }
+  };
+
+  return ApiWithoutInput({
+    name: 'Set3PCookie',
+    title: 'Set 3P Cookie',
+    onClick: setCookies,
+  });
+};
+
+const CheckCookieAccess = (): React.ReactElement => {
+  const refreshCookies = async (): Promise<string> => {
+    const access = await requestCookieAccess();
+    if (!access) {
+      return 'Permission denied. Either blocked by user or browser';
+    }
+
+    // Get and return the current cookie value
+    const cookieValue = document.cookie || 'No cookies found';
+    console.log('Read the document cookies:', cookieValue);
+    return cookieValue;
+  };
+
+  return ApiWithoutInput({
+    name: 'Check3PCookiesAccess',
+    title: 'Check 3P Cookies Access',
+    onClick: refreshCookies,
+  });
+};
+
+const hasCookieAccess = async (): Promise<boolean> => {
+  if (!document.requestStorageAccess) {
+    console.log(
+      'Storage Access API not supported. Assume it is an older browser that doesn not block 3P cookies and we have access.',
+    );
+    return true;
```

Comment:
oh, is it supposed to return true on storage access api not supported?

Reply:
Basically, if it is an older browser that does not block 3P cookies then we will have access and do not need to call Storage Access APIs to request access. We can use and set the document.cookie directly, for example, `document.cookie = 'testCookies=testvalue; path=/; Secure; SameSite=None';`. Please see the [Checking and requesting storage access](https://developer.mozilla.org/en-US/docs/Web/API/Storage_Access_API/Using#checking_and_requesting_storage_access) section in Using the Storage Access API document. I simplified the code to better align with the test need.